src/components/DataTable/index.js

```diff
@@ -203,31 +202,100 @@ const SbDataTable = {
     }
 
     const renderTable = () => {
+      let headerData = []
+      let bodyData = []
+
+      if (this.$slots.default) {
+        const children = this.$slots.default.filter((e) => e.tag)
+
+        headerData = children.map((element) => {
+          return {
+            ...element.componentOptions.propsData,
+          }
+        })
+
+        bodyData = this.sortedData.map((tableRow) => {
+          const columns = children.map((tableData) => {
+            return h(
+              tableData.componentOptions.Ctor,
+              {
+                ...tableData.data,
+                ...(tableData.componentOptions.listeners || {}),
+                props: {
+                  ...(tableData.data.props || {}),
+                  ...tableData.componentOptions.propsData,
+                  // passing the row property
+                  row: { ...tableRow },
+                },
+                attrs: {
+                  ...tableData.data.attrs,
+                },
+                on: tableData.componentOptions.listeners,
+              },
+              tableData.componentOptions.children
+            )
+          })
+
+          return h(
+            SbDataTableBodyRow,
+            {
+              props: {
+                allowSelection: this.allowSelection,
+                headers: [...headerData],
+                row: tableRow,
+                selectedRows: this.selectedRows,
+              },
+            },
+            columns
+          )
+        })
+      }
+
       return h(
         'table',
         {
           staticClass: 'sb-data-table__container',
         },
         [
-          !this.hideHeader &&
-            h(SbDataTableHeader, {
-              props: {
-                allowSelection: this.allowSelection,
-                allRowsSelected: this.allRowsSelected,
-                headers: this.headers,
-                selectedRowsLength: this.selectedRows.length,
-                selectionMode: this.selectionMode,
-                sortedKey: this.sortKey,
-              },
-            }),
-          h(SbDataTableBody, {
-            props: {
-              allowSelection: this.allowSelection,
-              headers: this.headers,
-              items: this.sortedData,
-              selectedRows: this.selectedRows,
-            },
-          }),
+          !this.hideHeader && this.headers.length
```

Comment:
What do you think to put this conditional statement into a computed, like `isShowHeader`?